Source/SwiftLintFramework/Rules/Style/DisallowTopLevelVariablesRule.swift

```diff
@@ -0,0 +1,64 @@
+import SwiftSyntax
+
+struct DisallowTopLevelVariablesRule: SwiftSyntaxRule, OptInRule, ConfigurationProviderRule {
+    var configuration: SeverityConfiguration = .init(.warning)
+
+    static let description: RuleDescription = .init(
+        identifier: "disallow_top_level_variables",
+        name: "Disallow Top-Level Variables",
+        description: "Top-Level variables and constants should be avoided",
+        kind: .style,
+        nonTriggeringExamples: [
+            Example(
+                """
+                struct A {
+                    public var a: Int = 12
+                }
+                """
+            )
+        ],
+        triggeringExamples: [
+            Example("var ↓a: Int = 11"),
+            Example("let ↓a: Int = 11"),
+            Example("private var ↓a: Int = 11"),
+            Example("public var ↓a: String?"),
+            Example("public var ↓b: Int { return 42 }")
+        ])
+
+    func makeVisitor(file: SwiftLintFile) -> ViolationsSyntaxVisitor {
+        Visitor()
+    }
+}
+
+private extension DisallowTopLevelVariablesRule {
+    final class Visitor: ViolationsSyntaxVisitor {
+        init() {
+            super.init(viewMode: .sourceAccurate)
+        }
```

Comment:
It's preferred to skip this constructor and initialize the visitor like `Visitor(viewMode: .sourceAccurate)` in line 29.